scripts/hooks/tech-lead-authoring-guard.py

```
@@ -0,0 +1,479 @@
+_PATH_TOKEN = r"['\"]?([^|;&\s<>'\"]+)['\"]?"
+_INTERPRETER_RE = r"(?:python|python3|node|ruby|perl|bash|sh|php|lua|Rscript)"
+
+
+def _extract_redirect_targets(command: str):
+    # `> FILE`, `>> FILE`, `1> FILE`, `2>&1 > FILE`, `&> FILE`, `&>> FILE`,
+    # `>| FILE`.
+    pattern = rf"(?:[0-9]+|&)?>>?\|?\s*{_PATH_TOKEN}"
+    return [m.group(1) for m in re.finditer(pattern, command)]
+
+
+def _extract_tee_targets(command: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_extract_tee_targets`

(ANN202)